packages/@icon-magic/generate/src/plugins/svg-to-custom-element.ts

```diff
@@ -62,6 +65,26 @@ export default function () {
 `;
 };
 
+/**
+ * To prepare the contents of svg for custom element.
+ *
+ * @param svgData Contents of svg
+ * @returns normalized contents of svg
+ */
+const normalizeSvgData = (svgData: string): string => {
+  // Parse XML from a string into a DOM Document
+  const doc = new DOMParser();
+  const xml = doc.parseFromString(svgData, 'image/svg+xml');
+  /* Strip id since it can cause id collisions if used multiple times
+     on a webpage. Instead use custom element name as a selector.
+   */
+  xml.documentElement.removeAttribute('id');
```

Comment:
I thought svg-generate will remove ids using svgo to start with. And the spaces too should be optimized after running through svgo

Reply:
Apparently, svg-generate does not remove ids. Instead it seems like it adds it as per [the code](https://jarvis.corp.linkedin.com/codesearch/result/?name=svg-generate.ts&path=icon-magic%2Fpackages%2F%40icon-magic%2Fgenerate%2Fsrc%2Fplugins&reponame=linkedin%2Ficon-magic#68). Also, seems like svg-generate does not remove the newlines in `<svg>`, hence, we need to strip them out in svg-to-cutom-element.